sympy/functions/elementary/piecewise.py

```diff
@@ -329,6 +329,8 @@ def _eval_derivative(self, x):
         return self.func(*[(diff(e, x), c) for e, c in self.args])
 
     def _eval_evalf(self, prec):
+        from mpmath.libmp.libmpf import prec_to_dps
+        prec = prec_to_dps(prec)
         return self.func(*[(e.evalf(prec), c) for e, c in self.args])
```

Comment:
Can't you instead just have this call `e._eval_evalf(prec)` instead of `e.evalf(prec)`?

Reply:
Tried it but shows error for NoneType error. 
![_eval_evalf](https://user-images.githubusercontent.com/30122295/41676188-69fe32c4-74e2-11e8-9813-41d24fff5816.png)

Reply:
The problem we were facing here was not due to wrong function calling but a generic OOP problem called The Fragile Base Class Problem.

Reply:
Here is a link which explains that problem quite well
https://medium.com/@cscalfani/goodbye-object-oriented-programming-a59cda4c0e53

Reply:
Ah my bad. It should be `e._evalf(prec)`. `_eval_evalf` is a lower level function which can return None. `_evalf` handles this (https://github.com/sympy/sympy/blob/91adc7462c1fadb7cc903e6282c12f2feac8234c/sympy/core/evalf.py#L1446)

The problem with your solution is that `prec_to_dps` can lose information, since dps is less fine-grained than prec (e.g., `prec_to_dps(53)` is the same as `prec_to_dps(54)`). 

I'm not really following the fragile base class thing. How does it apply here? I don't know Java well enough to follow the example in the article. Can you explain the difference between `a.add(element)` and `add(element)` inside a Java method?